nixos/default.nix

```diff
@@ -7,12 +16,18 @@ let
   eval = import ./lib/eval-config.nix {
     inherit system;
     modules = [ configuration ];
+    specialArgs = {
+      inherit lib;
+    };
```

Comment:
`eval-config.nix` takes a `lib` argument which you can use instead.

Reply:
Except that `lib` would then not propagate to the module arguments like you might expect. That argument is there as an optimization. In my lib overlays prototype I made that argument work as you expect, but this requires changes to `lib/modules.nix`, e.g. like this: `{ inherit config options; lib = args.lib or lib; } // specialArgs`.